src/ucp/core/ucp_request.inl

```diff
@@ -318,3 +293,98 @@ static UCS_F_ALWAYS_INLINE void ucp_request_send_tag_stat(ucp_request_t *req)
         UCP_EP_STAT_TAG_OP(req->send.ep, EAGER);
     }
 }
+
+static UCS_F_ALWAYS_INLINE void
+ucp_request_rndv_get_create(ucp_request_t *req)
+{
+    int i;
+    ucp_rndv_get_rkey_t *rkey;
+
+    ucs_trace_req("rendezvous-get create request %p", req);
+    req->send.rndv_get.rkey = ucs_mpool_get_inline(&(req->send.ep->worker)->rndv_get_mp);
+    req->flags |= UCP_REQUEST_FLAG_RNDV_MRAIL;
+
+    rkey = req->send.rndv_get.rkey;
+
+    for (i = 0; i < ucs_countof(rkey->rkey_bundle); i++) {
+        req->send.rndv_get.rkey->rkey_bundle[i].rkey = UCT_INVALID_RKEY;
+    }
+}
+
+static UCS_F_ALWAYS_INLINE
+uct_rkey_bundle_t *ucp_tag_rndv_rkey(ucp_request_t *req, int idx)
+{
+    return &req->send.rndv_get.rkey->rkey_bundle[idx];
+}
+
+static UCS_F_ALWAYS_INLINE void
+ucp_request_rndv_get_release(ucp_request_t *req)
+{
+    int i;
+    ucp_rndv_get_rkey_t *rkey;
+
+    ucs_trace_req("mrail release request %p", req);
+
+    if (req->send.rndv_get.rkey) {
+        rkey = req->send.rndv_get.rkey;
+        for (i = 0; i < ucs_countof(rkey->rkey_bundle); i++) {
+            if (ucp_tag_rndv_rkey(req, i)->rkey != UCT_INVALID_RKEY) {
+                uct_rkey_release(ucp_tag_rndv_rkey(req, i));
+            }
+        }
+    }
+
+    ucs_mpool_put_inline(req->send.rndv_get.rkey);
+    req->flags &= ~UCP_REQUEST_FLAG_RNDV_MRAIL;
+}
+
+static UCS_F_ALWAYS_INLINE void
+ucp_request_clear_rails(ucp_dt_state_t *state)
```

Comment:
should be in dt.h and named ucp_dt_... since it takes ucp_dt_state_t as argument and not request